lib/galaxy/datatypes/display_applications/parameters.py

```diff
@@ -107,19 +109,20 @@ def prepare(self, other_values, dataset_hash, user_hash, trans):
             # start conversion
             # FIXME: Much of this is copied (more than once...); should be some abstract method elsewhere called from here
             # find target ext
-            target_ext, converted_dataset = data.find_conversion_destination(self.formats, converter_safe=True)
-            if target_ext and not converted_dataset:
-                if isinstance(data, DisplayDataValueWrapper):
-                    data = data.value
-                new_data = next(iter(data.datatype.convert_dataset(trans, data, target_ext, return_output=True, visible=False).values()))
-                new_data.hid = data.hid
-                new_data.name = data.name
-                trans.sa_session.add(new_data)
-                assoc = trans.app.model.ImplicitlyConvertedDatasetAssociation(parent=data, file_type=target_ext, dataset=new_data, metadata_safe=False)
-                trans.sa_session.add(assoc)
-                trans.sa_session.flush()
-            elif converted_dataset and converted_dataset.state == converted_dataset.states.ERROR:
-                raise Exception("Dataset conversion failed for data parameter: %s" % self.name)
+            direct_match, target_ext, converted_dataset = data.find_conversion_destination(self.formats, converter_safe=True)
+            if direct_match:
```

Comment:
```suggestion
            if not direct_match:
```

Reply:
You are right. I amended the change to my previous commit